docs/decisions/broker-trigger.md

```diff
@@ -0,0 +1,166 @@
+# Broker and Trigger model (for 0.5.0)
+
+Decision date: 6 March 2019
+
+## Define High-Level Objects Related to Eventing
+
+The goal of eventing is to provide an abstract "bucket of events" which event
+consumers can sample from using filters on CloudEvent
+[context attributes](https://github.com/cloudevents/spec/blob/master/spec.md#context-attributes).
+Event producers can deliver events into the bucket without needing to understand
+the exact mechanics of routing to specific consumers. Event producers and
+consumers are decoupled temporally but also through a deliberately black-box
+abstraction which can optimize the underlying routing layer dynamically.
+
+## Definitions
+
+- **Event Producer**: a system which creates events based on occurrences. E.g.
+  GitHub (the hosted service).
+- **Event Consumer**: a destination which receives events. E.g. the application
+  processing an event. Typically, this is the final "sink" in the processing
```

Comment:
nit, do we need "final" here or can we drop it?

Reply:
Removed, thanks!